File: 1.0/src/cz/mg/vulkan/PFNvkGetSwapchainCounterEXT.java
package cz.mg.vulkan;

public class PFNvkGetSwapchainCounterEXT extends VkFunctionPointer {
    public PFNvkGetSwapchainCounterEXT() {
    }

    protected PFNvkGetSwapchainCounterEXT(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetSwapchainCounterEXT(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetSwapchainCounterEXT(long value) {
        setValue(value);
    }

    public PFNvkGetSwapchainCounterEXT(VkInstance instance) {
        super(instance, new VkString("vkGetSwapchainCounterEXT"));
    }

    public int call(VkDevice device, VkSwapchainKHR swapchain, int counter, VkUInt64 pCounterValue){
        return callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), swapchain != null ? swapchain.getVkAddress() : VkPointer.getNullAddressNative(), counter, pCounterValue != null ? pCounterValue.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callNative(long vkaddress, long device, long swapchain, int counter, long pCounterValue);

}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkGetSwapchainCounterEXT.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

jint Java_cz_mg_vulkan_PFNvkGetSwapchainCounterEXT_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong swapchain, jint counter, jlong pCounterValue){
    (void)env;
    (void)jc;
    PFN_vkGetSwapchainCounterEXT f = (PFN_vkGetSwapchainCounterEXT)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkSwapchainKHR*)jniLongToPointer(swapchain)),
        (VkSurfaceCounterFlagBitsEXT)counter,
        ((uint64_t*)jniLongToPointer(pCounterValue))
    );
}

